### PR TITLE
Homework 4 - Super team picker

### PR DESCRIPTION
**Pull request by harryji168**
add seed demo data, and backup pgsql using pg, 
add max, min value for input field

### CODE REVIEW

File: Super_Team_Picker/routes/cohorts.js [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+            .then(data => {
+                if(data===undefined){
+                    res.render('cohorts/show',{cohort:false,pageTitle: "Not Found - Super Team Picker"})
+                }else{
+                    let memberList = null;
+                    let isTeamCount = 'checked';
+                    let isMemberCount = null;                
+                    let members = data[0].members.split(',');
+                    let inputNumb = null;                         
+                    if (req.query.method && req.query.quantity) {
```
Comment (kaylyn94) [inline Super_Team_Picker/routes/cohorts.js:63]: to safe-guard that this operation will not fail, you can do `req.query && (req.query.method && req.query.quantity)`

File: Super_Team_Picker/routes/cohorts.js [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    if(isNaN(parseInt(req.params.id))){
+        res.render('cohorts/show',{cohort:false,pageTitle: "Not Found - Super Team Picker"});
+    }else{     
+        knex("cohorts")
+            .select("*")
+            .where({
+                id: req.params.id
+            })
+            .then(data => {
+                if(data===undefined){
```
Comment (kaylyn94) [inline Super_Team_Picker/routes/cohorts.js:55]: `!data`

File: Super_Team_Picker/routes/cohorts.js [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+        logoUrl: req.body.logoUrl
+    };
+    knex("cohorts").insert(cohortParmas).returning("id").then(data => {
+        res.redirect(`/cohorts/${data[0]}`);
+    })
+});
+ 
+
+router.get('/:id', (req, res) => {
+    if(isNaN(parseInt(req.params.id))){
```
Comment (kaylyn94) [inline Super_Team_Picker/routes/cohorts.js:46]: this is a huge blog of logic to put here. ideally, you may want to separate it out into a function and a function does one job and one job only.

**Comment by kaylyn94** [comment]
# SUPER TEAM PICKER HOMEWORK 4
## SCORESHEET FOR STUDENT Harry

### Create an Express app with Postgres to store cohorts. 
### Cohorts will have a list of members stored as comma separated names. 
### They'll also have a name and a logo. 

### GRADING CRITERIA:
### Course Value: 5%
### Assignment Value: 100%

### GRADE:
|Total Grade:||84|
|----------|----|----|
|Score| 84 | /100|
|Deductions| 00 |/20|

***LATE ASSIGNMENTS WILL BE DEDUCTED 5% OFF PER 24 HOURS, FOR A MAX OF 20% DEDUCTION OVERALL.
YOUR ASSIGNMENT IS CONSIDERED LATE 5 MINUTES AFTER THE DUE DATE AND TIME, UNLESS OTHERWISE DISCUSSED WITH YOUR INSTRUCTOR.***
__________________________________________________________
__________________________________________________________

### Part 1: Submitted and on Time

> Was the assignment complete and on time?

|Feature |Mark |Notes |
|--------|-----|------|
|Complete and submitted on time.|✅|All good!|
|On time, but incomplete.|||
|Complete, but submitted late.|||
|Incomplete and submitted late.|||
|Very late submission|||
|No submission|||

### Part 2: Features

> Was everything in assignment attempted?

|Feature |Mark |Notes |
|--------|-----|------|
|Express Project Setup |✅||
|The new page should be accessible from a link in the navbar |✅||
|The new page should show a form with all the fields necessary to create a cohort |✅||
|Submitting the form should store the cohort's logo image if there is one, then it should save the cohort's members, name and logoUrl to the cohorts table. |✅||
|It should redirect the user to the newly created cohort's show page. |✅||
|The Cohort page should list all cohorts with their name, members and a thumbnail of their logo. |✅||
|Each listed cohort should have a "Show Cohort" button that links to that cohort's show page. |✅||
|Showing a Single Cohort should show the cohort's name and members. |✅||
|The Show page should have a form to select the options for generating teams. |✅||
|Submitting the form should randomly generate a team according to the given options and display the team members with their numbers in a table below the form without redirecting the user to another page. |❌|Assigning members doesn't work correctly. Partial mark assigned.|
|The form should save its values even after it's been submitted. |✅||
|Add a button on the cohort's show page to delete a cohort. |✅||
|Add a button on a cohort's show page to edit a cohort. |✅||
|The edit button should send the user to the edit form where they can change the team members, team name, etc. |✅||

### Part 3: Styling and overall Impression

> Was Bootstrap installed and was attempts made to look like the mockups?

|Feature |Mark |Notes |
|--------|-----|------|
|Bootstrap installed and styling close or very close to mockups. |✅||
|Bootstrap installed and styling somewhat attempted to look like mockups or presentable. |||
|Bootstrap installed but no attempt for styling to look like mockups. |||
|No Bootstrap installed. |||
__________________________________________________________
__________________________________________________________

### ASSIGNMENT RUBRIC

|A+|A|A-|B+|B|B-|C+|C-F|
|--------|-------|-------|-------|-------|-------|-------|------|
|90%-100%|85%-89%|80%-84%|76%-79%|72%-75%|68%-71%|64%-67%|0%-63%|
|Assignment handed in on time and complete.  All instructions were followed perfectly.  All required features were created and works correctly.  All Stretches or Challenges were completed.  Styling and design of front end meets requirements, and extra creative effort has been applied to this assignment overall.|Assignment handed in on time and complete.  All instructions were followed perfectly.  All required features were created and works correctly.  Stretches or Challenges were attempted.  Styling and design of front end meets requirements, and looks presentable. **OR** Stretches or Challenges not attempted, but creative effort has been applied to the assignment overall.|Assignment handed in on time and complete.  All instructions were followed.  All required features were created and works correctly or mostly works.  No Stretches or Challenges were attempted.  Styling and design of front end meets requirements, and looks presentable.|Assignment handed in on time and complete.  All instructions were followed.  All required features were created or attempted.  No Stretches or Challenges were attempted.  Styling and design of front end meets requirements, and looks presentable.|Assignment handed in on time, but handed in as incomplete.  Most instructions were followed.  Most required features were created or attempted.  No Stretches or Challenges were attempted.  Styling and design of front end meets requirements.|Assignment handed in on time, but handed in as incomplete.  Instructions were somewhat followed.  Most required features were created or attempted.  No Stretches or Challenges were attempted.  Minimal effort in styling and design of front end.|Assignment handed in on time, but handed in as incomplete.  Instructions were somewhat followed.  Some required features were created or attempted.  No Stretches or Challenges were attempted.  Minimal effort in styling and design of front end.|Minimal or no effort.Incomplete/ missing components/insufficient.|

File: Super_Team_Picker/routes/cohorts.js (hunk 1 above)
Comment (harryji168) [reply Super_Team_Picker/routes/cohorts.js:63]: thanks

**Comment by harryji168** [comment]
could you give me a sample what wrong of "Assigning members doesn't work correctly. "  ?
thanks

Harry

**Comment by harryji168** [comment]
Hi, Kayleyn:

I don't know why "Assigning members doesn't work correctly", please give a sample?

thanks

Harry JI  604.253.3858 778.887.8868

________________________________

**Comment by harryji168** [comment]
hi, Kayln:

please check readme file , https://github.com/harryji168/Super_Team_Picker#readme with assignment member should work, I also did a lot of extra work for demo data and demo images
[Screenshot from 2021-10-10 21-52-34]

[https://user-images.githubusercontent.com/21187699/136735216-a0c82010-b366-4b16-b0ed-0a4466094081.png]

[https://opengraph.githubassets.com/b07004e14e42e6e6fa854b4220e82f9ba4b583a24c61c8a9401f7fa4ffcc6ea4/harryji168/Super_Team_Picker]<https://github.com/harryji168/Super_Team_Picker#readme>
GitHub - harryji168/Super_Team_Picker<https://github.com/harryji168/Super_Team_Picker#readme>
Contribute to harryji168/Super_Team_Picker development by creating an account on GitHub.
github.com


it should be work

thanks

Harry JI  604.253.3858 778.887.8868

________________________________

**Pull request merged**